### PR TITLE
feat(dataviz): add geo chart component

### PR DESCRIPTION
**Pull request by Kporal**
**What is the problem this PR is trying to solve?**

Extract geo chart from TDP into the dataviz shared component to be able to reuse it for TDS.

**Please check if the PR fulfills these requirements**

* [x] The PR commit message follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md)
* [x] Tests for the changes have been added (for bug fixes / features) And [non reg](./screenshots.md) done before need review
* [x] Docs have been added / updated (for bug fixes / features)
* [ ] Related design / discussions / pages (not in jira), if any, are all linked or available in the PR

**[ ] This PR introduces a breaking change**

### CODE REVIEW

File: packages/dataviz/package.json [hunk 1]
```diff
@@ -35,27 +35,41 @@
   },
   "dependencies": {
     "classnames": "^2.2.6",
-    "d3": "^6.3.0",
+    "d3-color": "^2.0.0",
```
Comment (jmfrancois) [inline packages/dataviz/package.json:38]: no please this goes against our cdn infra.
Please keep to use d3.
By the way there is a PR for that: https://github.com/Talend/ui-scripts/pull/176/files
But target was more for external package than internal.

**Review by jmfrancois — CHANGES_REQUESTED** [review]

File: packages/dataviz/package.json (hunk 1 above)
Comment (Kporal) [reply packages/dataviz/package.json]: Thanks for the help, wasn't aware of it. Did change it.

File: packages/dataviz/src/components/GeoChart/GeoChart.component.tsx [hunk 2]
```diff
@@ -0,0 +1,286 @@
...
+			.on('mouseover', d3Event => {
+				// if we have both label and entry key: "label (key): value", otherwise "label: value" or "key: value"
+				const tooltipLabel = `${label || entry.key}${
+					label && entry.key !== label ? ` (${entry.key})` : ''
+				}`;
+				setTooltipEntry({
+					key: tooltipLabel,
+					value: `${entry.value}`,
+				});
+				setTooltipPosition({
```
Comment (gjou-tlnd) [inline packages/dataviz/src/components/GeoChart/GeoChart.component.tsx:106]: I guess you could use the same state object for this

File: packages/dataviz/src/components/GeoChart/GeoChart.component.tsx [hunk 3]
```diff
@@ -0,0 +1,286 @@
...
+				</button>
+			</div>
+			<div ref={containerRef} />
+			{tooltipPosition && tooltipEntry && (
+				<div
+					ref={tooltipRef}
+					style={{
+						position: 'fixed',
+						top: `${tooltipPosition.y - (tooltipRef?.current?.offsetHeight ?? 0)}px`,
+						left: `${tooltipPosition.x - ((tooltipRef?.current?.offsetWidth ?? 0) / 2)}px`,
```
Comment (gjou-tlnd) [inline packages/dataviz/src/components/GeoChart/GeoChart.component.tsx:275]:
```suggestion
						top: `${tooltipPosition.y - (tooltipRef.current?.offsetHeight ?? 0)}px`,
						left: `${tooltipPosition.x - ((tooltipRef.current?.offsetWidth ?? 0) / 2)}px`,
```

File: packages/dataviz/src/components/GeoChart/GeoChart.scss [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+		}
+
+		&:active {
+			color: $regal-blue;
+		}
+	}
+
+	&__tooltip-value {
+		color: $scooter;
+	}
```
Comment (gjou-tlnd) [inline packages/dataviz/src/components/GeoChart/GeoChart.scss:68]:
```suggestion
```

**Comment by gjou-tlnd** [comment]
don't forget to update the changelog

File: packages/dataviz/src/components/GeoChart/GeoChart.component.tsx (hunk 2 above)
Comment (Kporal) [reply packages/dataviz/src/components/GeoChart/GeoChart.component.tsx:106]: Yes, of course, I find more readable that way thanks to explicit setters but it is fine to combine them into one

**Review by ncomont — APPROVED** [review]

**Review by gjou-tlnd — APPROVED** [review]

File: packages/dataviz/package.json [hunk 5]
```diff
@@ -35,27 +35,34 @@
...
   "dependencies": {
     "classnames": "^2.2.6",
-    "d3": "^6.3.0",
+    "d3": "^6.5.0",
     "date-fns": "^2.16.1",
-    "recharts": "^1.8.5"
+    "geojson": "^0.5.0",
+    "lodash": "^4.17.20",
+    "recharts": "^1.8.5",
+    "topojson-client": "^3.1.0"
```
Comment (jmfrancois) [inline packages/dataviz/package.json:43]: this dependencies seems dead: https://github.com/topojson/topojson-client

File: packages/dataviz/package.json [hunk 6]
```diff
@@ -35,27 +35,34 @@
   },
   "dependencies": {
     "classnames": "^2.2.6",
-    "d3": "^6.3.0",
+    "d3": "^6.5.0",
     "date-fns": "^2.16.1",
-    "recharts": "^1.8.5"
+    "geojson": "^0.5.0",
+    "lodash": "^4.17.20",
+    "recharts": "^1.8.5",
```
Comment (jmfrancois) [inline packages/dataviz/package.json:42]: do you think we can upgrade to recharts 2 ?
recharts 1 is based on a part of d3 5 so you have both d3 version today

**Review by jmfrancois — APPROVED** [review]

File: packages/dataviz/package.json (hunk 5 above)
Comment (Kporal) [reply packages/dataviz/package.json:43]: As discussed together, the library is stable and with very dependancies. So far, it is the one recommended no other alternatives. If a security issue raised concerning this lib then we'll tackle at the time needed.

File: packages/dataviz/package.json (hunk 6 above)
Comment (Kporal) [reply packages/dataviz/package.json:42]: Thanks for sharing this, no idea yet, we need to give a try and to evaluate the impact. This should be done in another dedicated PR to separate topics and to allow to move forward on smaller topics.

**Pull request merged**